app/plugins.js

```diff
@@ -228,46 +223,47 @@ function toDependencies(plugins) {
 }
 
 function install(fn) {
-  const {shell: cfgShell, npmRegistry} = exports.getDecoratedConfig();
+  const yarnPath = resolve(__dirname, '..', 'bin', 'yarn-standalone.js');
 
-  const shell = cfgShell && cfgShell !== '' ? cfgShell : undefined;
+  const env = {
+    NODE_ENV: 'production',
+    ELECTRON_RUN_AS_NODE: 'true'
+  };
 
-  shellEnv(shell).then(env => {
-    if (npmRegistry) {
-      env.NPM_CONFIG_REGISTRY = npmRegistry;
-    }
-    /* eslint-disable camelcase  */
-    env.npm_config_runtime = 'electron';
-    env.npm_config_target = process.versions.electron;
-    env.npm_config_disturl = 'https://atom.io/download/atom-shell';
-    /* eslint-enable camelcase  */
-    // Shell-specific installation commands
-    const installCommands = {
-      fish: 'npm prune; and npm install --production --no-shrinkwrap',
-      posix: 'npm prune && npm install --production --no-shrinkwrap'
-    };
-    // determine the shell we're running in
-    const whichShell = (typeof cfgShell === 'string' && cfgShell.match(/fish/)) ? 'fish' : 'posix';
-    const execOptions = {
-      cwd: path,
-      env
-    };
-
-    // https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback
-    // node.js requires command line parsing should be compatible with cmd.exe on Windows, should able to accept `/d /s /c`
-    // but most custom shell doesn't. Instead, falls back to default shell
-    if (process.platform !== 'win32') {
-      execOptions.shell = shell;
-    }
+  function yarn(args, cb) {
+    spawnQueue.push(end => {
+      const fullcmd = [process.execPath, yarnPath].concat(args).join(' ');
+      console.log('Launching yarn:', fullcmd);
+
+      cp.exec(fullcmd, {
+        cwd: path,
+        env,
```

Comment:
This production flag is for `yarn`. See https://yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-production
No side effect to Hyper.